portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,16 +54,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("viewer-comment");
-    String commentName = request.getParameter("viewer-comment-name");
-    String commentCombined = null;
-
     long timestamp = System.currentTimeMillis();
+    String name = null; 
+
+    if (userService.getCurrentUser()!=null){ //if someone is logged in 
+        Query query =
+            new Query("Comments")
+                .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, userService.getCurrentUser().getUserId()));
```

Comment:
nit: this line is pretty long, consider splitting up the query creation on multiple lines like
Internally at Google we have linters for java code that keep you at a 100 character line limit.